AK/FixedArray.h

```diff
@@ -15,40 +17,61 @@ namespace AK {
 template<typename T>
 class FixedArray {
 public:
-    FixedArray() = default;
-    explicit FixedArray(size_t size)
-        : m_size(size)
+    static FixedArray<T> empty()
     {
-        if (m_size != 0) {
-            m_elements = static_cast<T*>(kmalloc_array(m_size, sizeof(T)));
-            for (size_t i = 0; i < m_size; ++i)
-                new (&m_elements[i]) T();
-        }
+        return FixedArray<T>();
     }
-    ~FixedArray()
+
+    static ErrorOr<FixedArray<T>> try_create(size_t size)
     {
-        clear();
+        if (size == 0)
+            return FixedArray<T>();
+        T* elements = static_cast<T*>(kmalloc_array(size, sizeof(T)));
+        if (!elements)
+            return Error::from_errno(ENOMEM);
+        for (size_t i = 0; i < size; ++i)
+            new (&elements[i]) T();
+        return FixedArray<T>(size, elements);
+    }
+    static FixedArray<T> must_create_but_fixme_should_propagate_errors(size_t size)
```

Comment:
```suggestion
    }

    static FixedArray<T> must_create_but_fixme_should_propagate_errors(size_t size)
```

Reply:
I think it makes sense to group together functions that absolutely belong together.

Reply:
it looks ugly.

Reply:
Okay, then I will add a newline there.